src/cljs/aikakonematka/sound.cljs

```diff
@@ -0,0 +1,147 @@
+
+(def F5 (* 2 (frequencies-of-major-scale-in-4th-octave 3)))
+
+(def G5 (* 2 (frequencies-of-major-scale-in-4th-octave 4)))
+
+(defn play-beep! [sq-freq]
+  (run-with
+    (connect->
+      (square sq-freq)                                      ;Define the periodic waves
+      (low-pass 600)
+      (percussive 0.01 0.4)                                 ;Define how log it takes the note and a decay
+      (gain 0.3)                                            ;Alter amplitude of the wave                                       ;                                         ;
+      destination)
+    context
+    (current-time context)
+    1.0))
+
+(defn make-melody! []
+  (melody/phrase (:music-durations @util/game-state) (:music-pitches @util/game-state)))
+
+(def composition
+  (->>
+    (melody/phrase [1 1 0.67 0.33 1]                        ;The duration of each note
+                   [C5 C5 C5 D5 E5])                        ;The pitch of each note
+    (melody/then
+      (melody/phrase [0.67 0.33 0.67 0.33 1]
+                     [E5 D5 E5 F5 G5]))))
+
+(def composition2
+  (->>
+    (melody/phrase [1 1 0.67 0.33 1]
+                   [C5 C5 C5 D5 E5])
+    (melody/with
+      (melody/phrase [0.67 0.33 0.67 0.33 1]
+                     [E5 D5 E5 F5 G5]))))
+
+(def melody-box [composition composition2])
+
+(defn marimba [{:keys [pitch]}]
+  (connect->
+    (add (sine pitch) (sine (inc pitch)) (sine (* 2 pitch)))
+    (adshr 0.01 0.2 0.2 0.2 0.3)
+    (gain 0.1)))
+
+(defn bell
+  "An imitation bell, made by adding together harmonics."
+  [{:keys [pitch]}]
+  (let [harmonic (fn [n proportion]
+                   (connect->
+                     (sine (* n pitch))            ; Each harmonic is a sine wave.
+                     (percussive 0.01 proportion)  ; The attack and decay of each note.
+                     (gain 0.05)))]                ; Multiply the volume of each harmonic by 0.5.
+    (->> (map harmonic [1.0 2.0 3.0 4.1 5.2]       ; Each harmonic is a multiple of the base frequency.
+              [1.0 0.6 0.4 0.3 0.2])      ; Higher harmonics are weaker.
+         (apply add))))
+
+(defn organ [note]
+  (connect->
+    (add (sine (* 0.5 (:pitch note))) (triangle (:pitch note)))
+    (low-pass (* 4 (:pitch note)) (connect-> (sine 3) (gain 3)))
+    (adsr 0.1 0 1 0.3)
+    (gain 0.2)))
+
+(defn wah [{:keys [pitch]}]
+  (connect->
+    (sawtooth pitch)
+    (low-pass
+      (connect->
+        (constant (* 4 pitch))
+        (adsr 0.1 0.2 0.4 0.3)) 5)
+    (adsr 0.3 0.5 0.8 0.3)
+    (gain 0.1)))
+
+(def music-instrument [bell marimba organ wah])
+
+(defn play-from!
+  "Take a sequence of notes and play them from a set point in an audiocontext."
+  [audiocontext from notes]
+  (doseq [{:keys [time duration instrument] :as note} notes]
+    (let [at (+ time from)
+          synth-instance (-> note
+                             (dissoc :time)
+                             instrument)
```

Comment:
here instrument function is called, so map note should have a instrument key

Reply:
Gotcha! :)